src/app/components/FormInput/index.tsx

```diff
@@ -3,71 +3,55 @@ import React from 'react';
 import styles from './styles.module.scss';
 
 interface Props {
-  className?: string;
-  disabled?: boolean;
-  error?: string;
-  errorClassName?: string;
-  inputClassName?: string;
-  isTextarea?: boolean;
-  inputType: string;
-  label?: string | object;
-  labelClassName?: string;
-  name: string;
-  onBlur?: (e: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>) => void;
-  onChange: (e: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>) => void;
-  onFocus?: (e: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>) => void;
-  placeholder?: string;
-  readOnly?: boolean;
-  touched?: boolean;
-  submitCount?: number;
+  id: string;
+  label?: string;
+  type?: string;
+  validationSchema?: {
+    required: string;
+    maxLength: {
+      value: number;
+      message: string;
+    };
+    minLength: {
+      value: number;
+      message: string;
+    };
+    pattern: {
+      value: any;
+      message: string;
+    };
+  };
+  errors?: {
+    message: string;
+  };
+  isDirty?: boolean;
+  register: any;
+  onChange?: (e: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>) => void;
```

Comment:
I think we could improve the typing, but we can do it in another PR if you want.